src/darwin/Framework/CHIP/MTRDeviceType.mm

```diff
@@ -50,10 +50,37 @@ + (nullable MTRDeviceType *)deviceTypeForID:(NSNumber *)deviceTypeID
         return nil;
     }
 
-    return [[MTRDeviceType alloc]
-        initWithDeviceTypeID:deviceTypeID
-                        name:[NSString stringWithUTF8String:deviceTypeData->name]
-                   isUtility:(deviceTypeData->deviceClass != MTRDeviceTypeClass::Simple)];
+    auto * name = [[NSString alloc] initWithBytesNoCopy:(void *) deviceTypeData->name
```

Comment:
Is this no-copy here safe because initWithDeviceTypeID makes a copy?  If so, please document that.

Reply:
No, because our device type data structs point to string constants. Will add a comment.

Reply:
Actually I think I'll just change the code-gen template to store these a ObjC string constants directly.